components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/config/APIMConfigServiceImpl.java

```diff
@@ -382,96 +390,77 @@ public void addGAConfig(String organization, String gaConfig) throws APIManageme
         }
     }
 
-    @Override
-    public String getSelfSighupConfig(String organization) throws APIManagementException {
+    @Override public UserRegistrationConfigDTO getSelfSighupConfig(String organization) throws APIManagementException {
 
         if (organization == null) {
             organization = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
         }
         try {
             PrivilegedCarbonContext.startTenantFlow();
             PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(organization, true);
-            int tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
-            if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(organization)) {
-                APIUtil.loadTenantRegistry(tenantId);
-            }
-            UserRegistry registry = ServiceReferenceHolder.getInstance().getRegistryService()
-                    .getGovernanceSystemRegistry(tenantId);
-            if (registry.resourceExists(APIConstants.SELF_SIGN_UP_CONFIG_LOCATION)) {
-                Resource resource = registry.get(APIConstants.SELF_SIGN_UP_CONFIG_LOCATION);
-                return new String((byte[]) resource.getContent(), Charset.defaultCharset());
-            } else {
-                return null;
+            JSONObject tenantConfig = (JSONObject) new JSONParser().parse(getTenantConfig(organization));
+            if (tenantConfig.containsKey(APIConstants.SELF_SIGN_UP_NAME)) {
+                return getSignupConfigurationFromAdvancedConfigurations(
+                        (JSONObject) tenantConfig.get(APIConstants.SELF_SIGN_UP_NAME));
+            } else { // Following defaultConfig object will be used if the SelfSignUp configuration is not available in
+                // the Advanced tenant configuration
+                JSONObject defaultConfig = new JSONObject();
```

Comment:
just return   a new UserRegistrationConfigDTO() with signup disable

Reply:
It seems like this can create errors as we are not assigning any roles to carbon.super user.
What if we add the subscriber role to this defaultConfig as follows.

            _JSONObject tenantConfig = (JSONObject) new JSONParser().parse(getTenantConfig(organization));
            if (tenantConfig.containsKey(APIConstants.SELF_SIGN_UP_NAME)) {
                return getSignupConfigurationFromAdvancedConfigurations(
                        (JSONObject) tenantConfig.get(APIConstants.SELF_SIGN_UP_NAME));
            } else { // Following defaultConfig object will be used if the SelfSignUp configuration is not available in
                // the Advanced tenant configuration
                UserRegistrationConfigDTO defaultConfig = new UserRegistrationConfigDTO();
                defaultConfig.setSignUpEnabled(false);
                defaultConfig.getRoles().put("subscriber", false);
                return defaultConfig;
            }_